pkg/clusters/addons/kong/kong.go

```diff
@@ -296,3 +360,75 @@ func urlForService(ctx context.Context, cluster clusters.Cluster, nsn types.Name
 
 	return nil, fmt.Errorf("service %s has not yet been provisoned", service.Name)
 }
+
+func deployKongEnterpriseLicenseSecret(ctx context.Context, cluster clusters.Cluster, namespace, name string) error {
+	license := os.Getenv("KONG_ENTERPRISE_LICENSE")
+	if license == "" {
+		return fmt.Errorf("failed kong enterprise license from environment setting")
+	}
+
+	newSecret := &corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"license": []byte(license),
+		},
+	}
+
+	_, err := cluster.Client().CoreV1().Secrets(namespace).Create(ctx, newSecret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed creating kong-enterprise-license secret, err %w", err)
+	}
```

Comment:
This relates to above comments about the namespace creation not tolerating the namespace having already existed. Just a reminder here that you may want to consider tolerating the pre-existence of the license secret as well.

Reply:
As we agreed, in this test environment, we create namespace each time, user wont use deployed license secret. it is deployed using WithEnterpriseLicense(licenseJson)